File: kotlin-pr/50/kotlin/code-quality-naming/DataFilter.kt
class DataFilter {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    fun m1(l: List<Int>, r1: IntArray, r2: DoubleArray) {
        r1[0] = 0
        r2[0] = 0.0
        
        l.forEach { x ->
            r1[0] += x
        }
        
        r2[0] = if (l.isNotEmpty()) r1[0].toDouble() / l.size else 0.0
    }

    fun m2(v1: Int, v2: Double): String {
        return "T: $v1, A: ${"%.2f".format(v2)}"
    }
    // {/fact}
}

File: ruby-pr/53/ruby/code-quality-naming/data_filter.rb
class DataFilter
  # {fact rule=code-quality-naming@v1.0 defects=1}
  def m1(l, r1, r2)
    r1[0] = 0
    r2[0] = 0.0
    
    l.each do |x|
      r1[0] += x
    end
    
    r2[0] = l.length > 0 ? r1[0] / l.length : 0
  end

  def m2(v1, v2)
    "T: #{v1}, A: #{'%.2f' % v2}"
  end
  # {/fact}
end